src/main/java/javaslang/collection/Iterator.java

```diff
@@ -1679,6 +1667,22 @@ public T next() {
 
     abstract class AbstractIterator<T> implements Iterator<T> {
 
+        public boolean equals(Object o) {
```

Comment:
What is the use case for overriding Iterator.equals? If we compare it, it is gone.